### ISSUE TITLE
error to Deploying in vercel 

### ISSUE THREAD

**Issue by fediralydev** [magicuidesign/magicui#55]
![Uploading Screenshot 2024-05-24 at 7.33.51 PM.png…]()

**Comment by itsarghyadas** [comment]
@dillionverma could the issue be due to the difference between the Vercel Hobby and Pro plans?

@fediralydev you're on the Vercel free Hobby plan, correct?

**Comment by fediralydev** [comment]
Yeah i have just free plan amd i upload more projects and now problem , but this project i can't deploying, i try to create other account vercel for deploying this project same problem.

**Comment by itsarghyadas** [comment]
Yes, that might be the case. We'll look into it and make a fix so it can run on any plan.

Thanks for bringing this to our attention!

**Comment by fediralydev** [comment]
You're welcome! Please keep me updated on the progress and expedite the fix if possible. Thanks!

**Comment by itsarghyadas** [comment]
Absolutely. I'll update you as soon as we push the fix. Thanks.

**Comment by dillionverma** [comment]
Hi there! 

We just pushed a small fix to the main repo here

https://github.com/magicuidesign/magicui/commit/e8c07b8da2e1f6e04ae79b6240e086463a09b8ea

Can you try again and let me know if this reduces the size?

**Comment by fediralydev** [comment]
![Screenshot 2024-05-25 at 7 33 24 PM](https://github.com/magicuidesign/magicui/assets/109765255/c26a7385-622b-4c43-affd-a29b212fb630)
![Screenshot 2024-05-25 at 7 33 35 PM](https://github.com/magicuidesign/magicui/assets/109765255/a31d1e74-f52c-4b5e-86e0-3824ade05e52)

Same problem sir !

**Comment by fediralydev** [comment]
i think why you don't change api/og : route.tsx : 

const inter600 = fetch('https://your-storage-service.com/path/to/Inter-SemiBold.ttf').then((res) => res.arrayBuffer());
const inter700 = fetch('https://your-storage-service.com/path/to/Inter-Bold.ttf').then((res) => res.arrayBuffer());
const image = fetch('https://your-storage-service.com/path/to/og-bg-2.jpg').then((res) => res.arrayBuffer());
const logo = fetch('https://your-storage-service.com/path/to/icon.png').then((res) => res.arrayBuffer());

Applying the Changes
Update your api/og function as follows: 


import { capitalize } from "@/lib/utils";
import { ImageResponse } from "next/og";

export const runtime = "edge";

const inter600 = fetch('https://your-storage-service.com/path/to/Inter-SemiBold.ttf').then((res) => res.arrayBuffer());
const inter700 = fetch('https://your-storage-service.com/path/to/Inter-Bold.ttf').then((res) => res.arrayBuffer());
const image = fetch('https://your-storage-service.com/path/to/og-bg-2.jpg').then((res) => res.arrayBuffer());
const logo = fetch('https://your-storage-service.com/path/to/icon.png').then((res) => res.arrayBuffer());

export async function GET(req: Request) {
  const imageData = await image;
  const logoData = await logo;

  const url = new URL(req.url);
  const params = Object.fromEntries(url.searchParams);
  const title = capitalize(
    (params.title || "Magic UI")
      .replace(/-/g, " ")
      .split(" ")
      .slice(0, 3)
      .join(" ")
      .substring(0, 20),
  );

  try {
    return new ImageResponse(
      (
        <div tw="flex justify-center flex-col relative w-full h-full items-center bg-white">
          <img
            tw="absolute inset-0 w-full h-full"
            alt={title}
            // @ts-ignore
            src={imageData}
          />

          <h1 tw="absolute inset-0 flex justify-center items-center mb-57 flex-row">
            <img
              tw="h-10 w-10"
              alt={title}
              // @ts-ignore
              src={logoData}
            />
            <span
              tw="ml-3"
              style={{
                background: "linear-gradient(180deg, rgba(255,255,255,1) 0%, rgba(156, 163, 175, 1) 100%)",
                backgroundClip: "text",
                WebkitBackgroundClip: "text",
                color: "transparent",
                fontSize: 35,
                letterSpacing: "-1.5px",
                ...font("Inter 700"),
              }}
            >
              Magic UI
            </span>
          </h1>

          {title && (
            <p
              style={{
                position: "absolute",
                whiteSpace: "pre-wrap",
                background: "linear-gradient(315deg,#9E7AFF 0%,#FE8BBB 33.33%,#FFBD7A 66.67%,#F8EAC3 100%)",
                backgroundClip: "text",
                WebkitBackgroundClip: "text",
                textAlign: "center",
                fontSize: 100,
                letterSpacing: "-10px",
                color: "transparent",
                WebkitTextFillColor: "transparent",
                ...font("Inter 600"),
              }}
            >
              {title}
            </p>
          )}

          <h1 tw="absolute inset-0 flex justify-center items-center mt-57 flex-col">
            <span
              style={{
                background: "linear-gradient(180deg, rgba(255,255,255,1) 0%, rgba(156, 163, 175, 1) 100%)",
                backgroundClip: "text",
                WebkitBackgroundClip: "text",
                color: "transparent",
                fontSize: 25,
                letterSpacing: "-1.5px",
                ...font("Inter 700"),
              }}
            >
              Create magical landing pages in minutes.
            </span>
          </h1>
        </div>
      ),
      {
        width: 1200,
        height: 630,
        fonts: [
          {
            name: "Inter 600",
            data: await inter600,
          },
          {
            name: "Inter 700",
            data: await inter700,
          },
        ],
      },
    );
  } catch (e: any) {
    console.log(`${e.message}`);
    return new Response(`Failed to generate the image`, {
      status: 500,
    });
  }
}

function font(fontFamily: string) {
  return { fontFamily };
}

**Comment by fediralydev** [comment]
ihey , i fixed and its deployment succeeded in Vercel : **Update and Fix for Edge Function Size Limit Issue** 

![Screenshot 2024-05-25 at 11 06 04 PM](https://github.com/magicuidesign/magicui/assets/109765255/7f16df39-0765-4435-ab8a-186325e9271c)

We are pleased to announce that we have resolved the Edge Function size limit issue encountered during deployment on Vercel. The problem stemmed from the function exceeding the 1 MB size limit specified by our Vercel plan. To address this, we have updated our asset management approach by leveraging Cloudinary for hosting our fonts and images.

What Was Done:
Asset Hosting on Cloudinary:

Fonts, background images, and logos have been moved to Cloudinary.
This reduces the Edge Function size and ensures efficient asset delivery.
Code Update:

The URLs for the fonts, images, and logos have been updated in the code to point to Cloudinary-hosted assets.
Benefits:
Reduced Edge Function Size: By hosting assets externally on Cloudinary, we have significantly reduced the size of our Edge Functions, ensuring compliance with Vercel’s limits.
Improved Performance: Cloudinary’s robust CDN ensures faster asset delivery, enhancing the overall performance.
Scalability: This approach provides better scalability and management of our assets.
